aspnetcore/client-side/import-export-interop.md

```diff
@@ -0,0 +1,264 @@
+---
+title: Client-side development with .NET JavaScript `[JSImport]`/`[JSExport]` interop
+author: pavelsavara
+description: Learn how to perform JavaScript interop in client-side JavaScript apps using .NET JavaScript `[JSImport]`/`[JSExport]` interop.
+monikerRange: '>= aspnetcore-7.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 10/11/2022
+uid: client-side/import-export-interop
+---
+# Client-side development with .NET JavaScript `[JSImport]`/`[JSExport]` interop
+
+This article explains how perform JavaScript (JS) interop in client-side JS apps using .NET JS `[JSImport]`/`[JSExport]` interop API.
+
+Existing JS apps can use the expanded client-side WebAssembly support in .NET 7 to reuse .NET libraries from JS or to build novel .NET-based apps and frameworks.
+
+> [!NOTE]
+> This article focuses on JS interop in JS apps without any dependency on the [Blazor UI component model](xref:blazor/index). To distinguish Blazor WebAssembly from [client-side WebAssembly](https://webassembly.org), this article uses "*<:::no-loc text="WASM":::*" when referring to client-side WebAssembly.
+>
+> For guidance on using .NET JS `[JSImport]`/`[JSExport]` interop API in Blazor WebAssembly apps, see <xref:blazor/js-interop/import-export-interop>.
+>
+> Additional guidance for the scenarios covered by this article can be found in the [Use .NET from any JavaScript app in .NET 7](https://devblogs.microsoft.com/dotnet/use-net-7-from-any-javascript-app-in-net-7/) Microsoft .NET Blog post, including additional information for the following subjects:
+>
+> * Information on the *Port of famous Todo-MVC to .NET on WASM* sample app based on [TodoMVC](https://todomvc.com/) and [Playwright for .NET](https://playwright.dev/dotnet/). For more information, see the [Additional example](#additional-example) section later in this article.
+> * App optimization.
+> * Performance.
+
+These approaches are appropriate when you only expect to run on :::no-loc text="WASM":::. Libraries can make a runtime check to determine if the app is running on :::no-loc text="WASM"::: by calling <xref:System.OperatingSystem.IsBrowser%2A?displayProperty=nameWithType>.
+
+## Prerequisites
+
+[!INCLUDE[](~/includes/7.0-SDK.md)]
+
+## Namespace
+
+.NET JS `[JSImport]`/`[JSExport]` interop API is controlled by attributes from the `System.Runtime.InteropServices.JavaScript` namespace.
+
+## Experimental workload and project templates
```

Comment:
I expect that most users looking to integrate .NET into JS code will be doing so from existing JS based apps and not starting from scratch. We should document this feature from that perspective. So instead of leading with these experimental templates I think we should describe how you manually use .NET from JS. At the end of this article we can mention the templates as a reference point.

Reply:
I expect it will take some time to work through these details. Let's separate out this new Blazor-less article into a separate PR so we can proceed with the Blazor based article.